maps/src/main/java/com/gluonhq/impl/maps/TileImageView.java

```diff
@@ -68,12 +69,16 @@ public TileImageView(int zoom, long i, long j) {
             }).thenAccept(file -> {
                 logger.fine("Tile from downloaded file " + zoom + "/" + i + "/" + j);
                 downloading.setValue(false);
-                setImage(file);
+                Platform.runLater(() -> {
```

Comment:
The block in `future.thenAccept(file -> {})` runs outside the JavaFX Application thread, and since all three calls inside of it are setters of JavaFX properties (`downloading`, `imageProperty` and `progress`), I'd say it is better to extend `Platform.runLater()` to all three of them.